src/RetryRequest.jl

```diff
@@ -7,6 +7,13 @@ using ..MessageRequest
 using ..Messages
 import ..@debug, ..DEBUG_LEVEL
 
+@static if isdefined(Base, :DNSError)
```

Comment:
I would rather this be based on `VERSION < ` so that it stands out more as a compat hack that should be removed later (as opposed to being some kind of cleverness used to dynamically handle optional dependancies or some such). I've found recently when I come to review compat stuff that is done based on isdefined or isapplicable that it takes some time to figure out which is the old way and which is the new way. An explicit `VERSION < ` is ambiguous.

Reply:
Meh, this is such a minor compat case and I don't want to go digging through commit history to find what version it was moved out on. This "autodetects" the compat based on actual, current definitions and we'll just take it out in a few weeks anyway when 0.7 is released.

Reply:
> don't want to go digging through commit history to find what version it was moved out on

I'm not suggesting that.

We should defined one version of 0.7 that we support and use that for all the compat decisions: 
https://github.com/JuliaWeb/HTTP.jl/blob/a1c6654fcb45f342bf7152b6aa4c7bc5aa25a832/src/compat.jl#L3

e.g. If you've just made things work on 0.7, update `v07 =` to whatever `VERSION` says at the time.

Reply:
addressed